src/server/snapshot.cc

```diff
@@ -175,13 +177,36 @@ bool SliceSnapshot::FlushSfile(bool force) {
   }
   VLOG(2) << "FlushSfile " << sfile_->val.size() << " bytes";
 
-  DbRecord rec = GetDbRecord(savecb_current_db_, std::move(sfile_->val), num_records_in_blob_);
+  uint32_t record_num = num_records_in_blob_;
   num_records_in_blob_ = 0;  // We can not move this line after the push, because Push is blocking.
-  dest_->Push(std::move(rec));
+  PushFileToChannel(sfile_.get(), savecb_current_db_, record_num, enhanced_compression_);
 
   return true;
 }
 
+string_view ZstdCompress::Compress(string_view str) {
+  size_t buf_size = ZSTD_compressBound(str.size());
+  if (compr_buf_.size() < buf_size) {
+    compr_buf_.reserve(buf_size);
+  }
+  size_t compressed_size =
+      ZSTD_compressCCtx(cctx_, compr_buf_.data(), compr_buf_.capacity(), str.data(), str.size(), 1);
```

Comment:
please introduce a flag `snapshot_compression_level` - it will be convenient for playing with compression.